bundles/org.eclipse.e4.ui.workbench/src/org/eclipse/e4/ui/internal/workbench/PartServiceImpl.java

```diff
@@ -696,9 +696,6 @@ private void activate(MPart part, boolean requiresFocus, boolean activateBranch)
 			activePart = part;
 			return;
 		}
-		if (partActivating == part) { // avoid recursive activations. (e.g. from focusGui)
```

Comment:
can you add modelService.bringToTop(part) prior to returning instead of removing the code?

Reply:
yes, it works

Reply:
can we add modelService.bringToTop(part) statement after UIEvents.publishEvent(UIEvents.UILifeCycle.ACTIVATE, part) instead?

Reply:
no, with this changes this is not working

Reply:
since the recursion is from focusGui, maybe we should add modelService.bringToTop(part) statement prior to focusGui.

Reply:
bringToTop has a side-effect of activating Inventory.

we can't avoid that side-effect easily.

maybe we should disallow any recursion.  i.e. return if partActivating is not null.